### PR TITLE
Core: Add animation kit

### PR DESCRIPTION
**Pull request by ghost**
**Changes proposed:**
-  Adds aiAnimKit, movementAnimKit, & meleeAnimKit fields to creature_addon.

**Target branch(es):** 6.x

**Tests performed:** (It build, worked in-game)

### CODE REVIEW

File: sql/updates/world/6.x/2016_08_20_01_world.sql [hunk 1]
```diff
@@ -0,0 +1,6 @@
+ALTER TABLE `creature_addon` ADD `aiAnimKit` SMALLINT NOT NULL DEFAULT 0 AFTER `emote`;
```
Comment (Vincent-Michael) [inline sql/updates/world/6.x/2016_08_20_01_world.sql]: optimize your sql query

Comment (Shauren) [inline sql/updates/world/6.x/2016_08_20_01_world.sql]: wtf do you want? this is only adding fields

Comment (Vincent-Michael) [inline sql/updates/world/6.x/2016_08_20_01_world.sql]: only 1x ALTER use

File: src/server/game/Globals/ObjectMgr.cpp [hunk 2]
```diff
@@ -608,6 +611,27 @@ void ObjectMgr::LoadCreatureTemplateAddons()
             creatureAddon.emote = 0;
         }
 
+        if (!sAnimKitStore.LookupEntry(creatureAddon.aiAnimKit))
```
Comment (DDuarte) [inline src/server/game/Globals/ObjectMgr.cpp]: `if (creatureAddon.aiAnimKit && !sAnimKitStore.LookupEntry(creatureAddon.aiAnimKit))`

Should only be checked if not 0 (and then the if inside this one is not required)

File: src/server/game/Entities/Creature/Creature.h [hunk 3]
```diff
@@ -358,6 +358,9 @@ struct CreatureAddon
     uint32 bytes1;
     uint32 bytes2;
     uint32 emote;
+    uint16 aiAnimKit;
+    uint16 movementAnimKit;
+    uint16 meleeAnimKit;
```
Comment (Vincent-Michael) [inline src/server/game/Entities/Creature/Creature.h]: initialize your stuff

File: sql/updates/world/6.x/2016_08_20_01_world.sql (hunk 1 above)
Comment (malcrom) [inline sql/updates/world/6.x/2016_08_20_01_world.sql]: ?

Comment (Vincent-Michael) [inline sql/updates/world/6.x/2016_08_20_01_world.sql]: https://gist.github.com/Vincent-Michael/6c3a4506696b919f7a4d8334bd3ee035

**Comment by DDuarte** [comment]
LGTM

**Comment by Sar777** [comment]
why is it put where it does not belong?

**Comment by DDuarte** [comment]
What?

**Pull request merged**